sympy/physics/control/lti.py

```diff
@@ -1422,3 +1445,551 @@ def _eval_rewrite_as_TransferFunction(self, num, den, **kwargs):
 
     def __neg__(self):
         return Feedback(-self.num, self.den)
+
+
+def _to_TFM(mat, var):
+    """Private method to convert ImmutableMatrix to TransferFunctionMatrix efficiently"""
+    def group(flat, size):
+        return [flat[i:i+size] for i in range(0, len(flat), size)]
+    tfm_cols = mat.args[1]
+    flat_list = list(mat.args[2])
+    flat_list = [TransferFunction.from_rational_expression(elem, var) for elem in flat_list]
+    arg = group(flat_list, tfm_cols)
```

Comment:
Don't access `args` directly like this. You should not presume the layout of the args of a class from outside. Instead the class should provide accessors like `mat.cols`.

In any case this code can just be:
```python
to_tf = lambda expr: TransferFunction.from_rational_expression(expr, var)
arg = [[to_tf(expr) for expr in row] for row in mat.tolist()]
```